### PR TITLE
api v0.3 -> v0.4 (pagination)

### PR DESCRIPTION
**Pull request by dimaninc**

### CODE REVIEW

**Comment by dmitry-mukhin** [comment]
tests :(

**Comment by dmitry-mukhin** [comment]
please add file iteration method, so it's easy to list all files in a project, respecting filters

File: src/Uploadcare/Api.php [hunk 1]
```diff
@@ -111,26 +123,136 @@ public function getPublicKey()
...
+
+  /**
+   * Return an iterator of File objects to work with.
+   *
+   * This class provides iteration over all uploaded files. You can specify:
+   *   - $options['from'] - a DateTime object or string from which objects will be iterated;
+   *   - $options['to'] - a DateTime object or string to which objects will be iterated;
+   *   - $options['limit'] - a total number of objects to be iterated;
+   *   - $options['request_limit'] - a number of objects to be downloaded per request.
+   *     If not specified, all available objects are iterated;
```
Comment (dmitry-mukhin) [inline src/Uploadcare/Api.php]: this is for previous `limit` option, right?

**Comment by dmitry-mukhin** [comment]
rename "portion" to "chunk"? seems to be more broadly used term for these kind of things.

**Comment by dmitry-mukhin** [comment]
Can you also provide two example scripts using this:

`remove.php` - should delete all files older than X days
`download.php` - should download all files in a project to local disc

It may or may not be done as a separate PR.

File: src/Uploadcare/Api.php (hunk 1 above)
Comment (dimaninc) [inline src/Uploadcare/Api.php]: sure, fixed that in new commit

**Comment by dimaninc** [comment]
renamed portion to chunk
`remove.php` and `download.php` would be in a new branch

**Comment by dimaninc** [comment]
bump
merged it all, added comments for some methods

**Comment by dmitry-mukhin** [comment]
bump version (1.5.0) and add something to changelog

**Comment by dimaninc** [comment]
bumped version, added changes to changelog

**Pull request merged**